pkg/apis/application/v1alpha1/types.go

```diff
@@ -251,6 +256,11 @@ func (a *ApplicationSource) AllowsConcurrentProcessing() bool {
 	return true
 }
 
+// IsRef returns true when the application source is of type Ref
+func (a *ApplicationSource) IsRef() bool {
+	return a.Ref != ""
+}
```

Comment:
Is this function being used anywhere? I could not find it to be referenced anywhere.

Reply:
nice catch! It was used in a previous version that I did, but now it's not used

Reply:
In any case, I have seen some places to use it and align the code with other checks like IsHelm, so I'll use it. WDYT?

Reply:
If we can replace `ref != nil` everywhere, it would be good to keep it.